c10/core/ScalarType.h

```diff
@@ -109,7 +109,8 @@ namespace c10 {
   _(double, Double, d)                        \
   _(c10::qint8, QInt8, i)                     \
   _(c10::quint8, QUInt8, i)                   \
-  _(c10::qint32, QInt32, i)
+  _(c10::qint32, QInt32, i)                   \
+  _(bool, Bool, i)
```

Comment:
this doesn't seem consistent (some of the FORALLs have bool and some dont).  What makes this special?

Reply:
As of right now, there are only 2 FORALL marcos that dont have Bool in it.
1) AT_FORALL_SCALAR_TYPES_EXCEPT_QINT
Its being used only in core/Scalar.h for conversion and there is a bool overload because of a compiler error. Please see a comment [there](https://github.com/pytorch/pytorch/blob/master/c10/core/Scalar.h#L38). 

2) AT_FORALL_SCALAR_TYPES_EXCEPT_HALF_AND_QINT()
I didn't touch it as there was never a real need to. its being used for in
- THBlasUtils.h 
- TensorFactories.cpp

In order to keep this PR clean, i created an [issue](https://github.com/pytorch/pytorch/issues/22744) to figure out what to do with the AT_FORALL_SCALAR_TYPES_EXCEPT_HALF_AND_QINT macro.